rocAL/rocAL/source/pipeline/tensor.cpp

```diff
@@ -77,6 +82,22 @@ vx_enum interpret_tensor_data_type(RocalTensorDataType data_type) {
     }
 }
 
+void allocate_host_or_pinned_mem(void **ptr, size_t size, RocalMemType mem_type) {
+    if (mem_type == RocalMemType::HIP) {
+#if ENABLE_HIP
+        hipError_t err = hipHostMalloc((void **)ptr, size, hipHostMallocDefault);
+        if(err != hipSuccess || !*ptr)
+            THROW("hipHostMalloc of size " + TOSTR(size) + " failed " + TOSTR(err))
+        err = hipMemset((void *)*ptr, 0, size);
+        if(err != hipSuccess)
+            THROW("hipMemset of size " + TOSTR(size) + " failed " + TOSTR(err))
+#endif
+    } else {
```

Comment:
The ENABLE_HIP is enabled only if CMake is built with HIP backend otherwise we will have only HOST backend in that case we cannot use the HIP API for HOST allocations